cmd/thanos/store.go

```diff
@@ -114,6 +116,12 @@ func (sc *storeConfig) registerFlag(cmd extkingpin.FlagClause) {
 
 	sc.objStoreConfig = *extkingpin.RegisterCommonObjStoreFlags(cmd, "", true)
 
+	cmd.Flag("initial-sync-retry-interval", "Interval between retries for the initial sync of the bucket store.").
+		Default("2s").DurationVar(&sc.initialSyncRetryInterval)
+
+	cmd.Flag("initial-sync-retry-timeout", "Timeout for the initial sync of the bucket store.").
+		Default("3m").DurationVar(&sc.initialSyncRetryTimeout)
+
```

Comment:
I wonder if this will not break users - e.g. if initial sync is failing on config error and you rely pod to crashloop for that.. you just make this process much longer 🤔 


Potentially keeping it smaller like 30s for default and then you can play with lager value makes sense